### PR TITLE
[PBW-5959] - Fixed problematic for..in loop

### PR DESCRIPTION
**Pull request by jpfranco**

### CODE REVIEW

File: js/framework/AnalyticsConstants.js [hunk 1]
```diff
@@ -561,11 +561,10 @@ if (!OO.Analytics.EVENT_DATA)
...
     var checkBitrateProfileList = OO._.bind(checkDataType, this, "VideoBitrateProfileLookupData");
-    var list = checkBitrateProfileList(bitrateProfileArray, "bitrateProfileArray", ["array"]);
+    var list = checkBitrateProfileList(bitrateProfileArray, "bitrateProfileArray", ["array"]) || [];
     this.profiles = {};
-    for(var key in list)
-    {
-      var entry = list[key];
+    for (var i = 0; i < list.length; i++) {
+      var entry = list[i];
       this.profiles[entry.id] = entry;
```
Comment (pilievOoyala) [inline js/framework/AnalyticsConstants.js]: can you add a null check for entry ?

File: js/framework/AnalyticsConstants.js [hunk 2]
```diff
@@ -561,12 +561,13 @@ if (!OO.Analytics.EVENT_DATA)
...
   {
     var checkBitrateProfileList = OO._.bind(checkDataType, this, "VideoBitrateProfileLookupData");
-    var list = checkBitrateProfileList(bitrateProfileArray, "bitrateProfileArray", ["array"]);
+    var list = checkBitrateProfileList(bitrateProfileArray, "bitrateProfileArray", ["array"]) || [];
     this.profiles = {};
-    for(var key in list)
-    {
-      var entry = list[key];
-      this.profiles[entry.id] = entry;
+    for (var i = 0; i < list.length; i++) {
```
Comment (dchauOoyala) [inline js/framework/AnalyticsConstants.js]: Sorry to nitpick, but can you please place the `{`'s on a newline. We have been trying to adopt this style for our new code.

Comment (mityaha) [inline js/framework/AnalyticsConstants.js]: This is something new - we always have { on the same line, Doug

Comment (jpfranco) [inline js/framework/AnalyticsConstants.js]: I like it on the same line, but I changed it for consistency with the rest of the code

Comment (mityaha) [inline js/framework/AnalyticsConstants.js]: If this is the style of all the other code in this file, then it's OK

Comment (jpfranco) [inline js/framework/AnalyticsConstants.js]: Damn, I missed the one on the if statement. I'll change that as well.

**Comment by pilievOoyala** [comment]
Looks good!  Thanks!

**Pull request merged**